### PR TITLE
Replace use of `relref` shortcode with `docs/reference` so links work in Grafana Cloud

### PR DESCRIPTION
**Pull request by jdbaldry**
Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

### CODE REVIEW

File: docs/sources/dashboards/build-dashboards/create-dashboard/index.md [hunk 1]
```diff
@@ -135,6 +135,9 @@ You can size a dashboard panel to suits your needs.
 1. To adjust the size of the panel, click and drag the lower-right corner of the panel.
 
 {{% docs/reference %}}
+[Dashboard]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/datasources#special-data-sources"
+[Dashboard]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/datasources#special-data-sources"
```
Comment (imatwawana) [inline docs/sources/dashboards/build-dashboards/create-dashboard/index.md:139]: I don't understand why this (and all the other ones) don't point to Cloud docs...
```suggestion
[Dashboard]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/datasources#special-data-sources"
```

**Review by imatwawana — APPROVED** [review]
The links work, but when I test these links in my local from Cloud, I'm sent back to the Grafana docs. I'm not sure if that's the intended behaviour here. I'm approving because this is a bigger question than just this PR.

File: docs/sources/panels-visualizations/configure-data-links/index.md [hunk 2]
```diff
@@ -133,6 +133,12 @@ When creating or updating a data link, press Cmd+Space or Ctrl+Space on your key
 1. Click **Save** in the upper right to save your changes to the dashboard.
 
 {{% docs/reference %}}
+[Cloudwatch]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/datasources/aws-cloudwatch/query-editor#deep-link-grafana-panels-to-the-cloudwatch-console-1"
+[Cloudwatch]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/datasources/aws-cloudwatch/query-editor#deep-link-grafana-panels-to-the-cloudwatch-console-1"
```
Comment (jdbaldry) [inline docs/sources/panels-visualizations/configure-data-links/index.md:137]:
```suggestion
[Cloudwatch]: "/docs/grafana-cloud/ -> /docs/grafana-cloud/connect-externally-hosted/data-sources/aws-cloudwatch/query-editor#deep-link-grafana-panels-to-the-cloudwatch-console-1"
```

File: docs/sources/panels-visualizations/configure-data-links/index.md [hunk 3]
```diff
@@ -133,6 +133,12 @@ When creating or updating a data link, press Cmd+Space or Ctrl+Space on your key
...
+
+[Google Cloud Monitoring]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/datasources/google-cloud-monitoring/query-editor#deep-link-from-grafana-panels-to-the-google-cloud-console-metrics-explorer"
+[Google Cloud Monitoring]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/datasources/google-cloud-monitoring/query-editor#deep-link-from-grafana-panels-to-the-google-cloud-console-metrics-explorer"
```
Comment (jdbaldry) [inline docs/sources/panels-visualizations/configure-data-links/index.md:140]:
```suggestion
[Google Cloud Monitoring]: "/docs/grafana-cloud/ -> /docs/grafana-cloud/connect-externally-hosted/data-sources/google-cloud-monitoring/query-editor#deep-link-from-grafana-panels-to-the-google-cloud-console-metrics-explorer"
```

File: docs/sources/panels-visualizations/visualizations/datagrid/index.md [hunk 4]
```diff
@@ -121,3 +121,8 @@ To delete or clear multiple rows, take the following steps:
 The same rules apply to columns by clicking the column headers.
 
 To delete all rows, use the "select all" checkbox at the top left corner of the datagrid. This selects all rows and allows you to delete them using the context menu.
+
+{{% docs/reference %}}
+[special data sources]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/datasources#special-data-sources"
+[special data sources]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/datasources#special-data-sources"
```
Comment (jdbaldry) [inline docs/sources/panels-visualizations/visualizations/datagrid/index.md:127]:
```suggestion
[special data sources]: "/docs/grafana-cloud/ -> /docs/grafana-cloud/connect-externally-hosted/data-sources#special-data-sources"
```

**Pull request merged**